x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/FollowIndexAction.java

```diff
@@ -275,24 +276,43 @@ void finalizeResponse() {
         }
     }
 
-
     static void validate(IndexMetaData leaderIndex, IndexMetaData followIndex, Request request) {
         if (leaderIndex == null) {
             throw new IllegalArgumentException("leader index [" + request.leaderIndex + "] does not exist");
         }
-
         if (followIndex == null) {
             throw new IllegalArgumentException("follow index [" + request.followIndex + "] does not exist");
         }
         if (leaderIndex.getSettings().getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), false) == false) {
             throw new IllegalArgumentException("leader index [" + request.leaderIndex + "] does not have soft deletes enabled");
         }
-
         if (leaderIndex.getNumberOfShards() != followIndex.getNumberOfShards()) {
             throw new IllegalArgumentException("leader index primary shards [" + leaderIndex.getNumberOfShards() +
                 "] does not match with the number of shards of the follow index [" + followIndex.getNumberOfShards() + "]");
         }
-        // TODO: other validation checks
+        if (leaderIndex.getRoutingNumShards() != followIndex.getRoutingNumShards()) {
+            throw new IllegalArgumentException("leader index number_of_routing_shards [" + leaderIndex.getRoutingNumShards() +
+                "] does not match with the number_of_routing_shards of the follow index [" + followIndex.getRoutingNumShards() + "]");
+        }
+        if (leaderIndex.getState() != IndexMetaData.State.OPEN || followIndex.getState() != IndexMetaData.State.OPEN) {
+            throw new IllegalArgumentException("leader and follow index must be open");
+        }
+        Map<String, Object> leaderMapping = getMapping(leaderIndex);
+        Map<String, Object> followerMapping = getMapping(followIndex);
+        if (leaderMapping.equals(followerMapping) == false) {
+            throw new IllegalArgumentException("the leader and follower mappings must be identical");
+        }
+        Map<String, Settings> leaderAnalysisSettings = leaderIndex.getSettings().getGroups("index.analysis");
+        Map<String, Settings> followerAnalysisSettings = followIndex.getSettings().getGroups("index.analysis");
```

Comment:
I wonder if we should flip this and only while list things that can be different (like number of replicas, allocation filtering etc.). Thoughts?

Reply:
This does seem safer, however almost all settings are allowed to be different so whitelist is going to be very large then.

Reply:
This is indeed what I wasn't sure about. I did a very brief random sample of IndexSetting and I'm not sure we need to white list so many settings when compared to all the settings that are in there?

Reply:
So you mean that we only whitelist a few settings (like refresh_rate) while we could whitelist more settings (like translog settings) but since there are not commonly used we just not allow it?

Reply:
to be clear - by not whitelisting them it doesn't mean people can't change them - it means they need to be same at both source and target. I think things like refresh rate and and translog settings are ok in that regard?

Reply:
> to be clear - by not whitelisting them it doesn't mean people can't change them - it means they need to be same at both source and target.

Yes.

> I think things like refresh rate and and translog settings are ok in that regard?

What settings do you think are likely to be different in leader and follower index?

Reply:
@bleskes I've added a whitelist of allowed settings that may be different: 38f2cf188cfd119bd178d8d1bceff5957f30dec0